controller/src/main/java/org/apache/rocketmq/controller/impl/manager/ReplicasInfoManager.java

```diff
@@ -316,6 +320,41 @@ public ControllerResult<Void> getSyncStateData(final List<String> brokerNames) {
         return result;
     }
 
+    public ControllerResult<Void> cleanBrokerData(final CleanControllerBrokerDataRequestHeader requestHeader,
+        final BiPredicate<String, String> brokerAlivePredicate) {
+        final ControllerResult<Void> result = new ControllerResult<>();
+
+        final String clusterName = requestHeader.getClusterName();
+        final String brokerName = requestHeader.getBrokerName();
+        final String brokerAddrs = requestHeader.getBrokerAddress();
+
+        //if SyncStateInfo.masterAddress is not empty, at least one broker with the same BrokerName is alive
+        SyncStateInfo syncStateInfo = this.syncStateSetInfoTable.get(brokerName);
+        if (StringUtils.isBlank(brokerAddrs) && null != syncStateInfo && StringUtils.isNotEmpty(syncStateInfo.getMasterAddress())) {
+            String remark = String.format("Broker %s is still alive, clean up failure", requestHeader.getBrokerName());
+            result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, remark);
+            return result;
+        }
+        Set<String> brokerAddressSet = null;
+        if (StringUtils.isNotBlank(brokerAddrs)) {
+            brokerAddressSet = Stream.of(brokerAddrs.split(";")).collect(Collectors.toSet());
+            for (String brokerAddr : brokerAddressSet) {
+                if (brokerAlivePredicate.test(clusterName, brokerAddr)) {
+                    String remark = String.format("Broker [%s,  %s] is still alive, clean up failure", requestHeader.getBrokerName(), brokerAddr);
+                    result.setCodeAndRemark(ResponseCode.CONTROLLER_INVALID_REQUEST, remark);
+                    return result;
+                }
+            }
+        }
```

Comment:
Hi, @mxsm I don't think it is necessary to check whether the brokers are alive before cleaning. Even if the living brokers are cleaned, they will register again.

Reply:
I got it